server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/AllocationBalancingRoundMetrics.java

```diff
@@ -0,0 +1,113 @@
+    private final LongCounter shardMovesCounter;
+    private final LongHistogram shardMovesHistogram;
+
+    private final LongHistogram shardCountHistogram;
+    private final DoubleHistogram diskUsageHistogram;
+    private final DoubleHistogram writeLoadHistogram;
+    private final DoubleHistogram totalWeightHistogram;
+
+    public static AllocationBalancingRoundMetrics NOOP = new AllocationBalancingRoundMetrics(MeterRegistry.NOOP);
+
+    public AllocationBalancingRoundMetrics(MeterRegistry meterRegistry) {
+        this.balancingRoundCounter = meterRegistry.registerLongCounter(
+            NUMBER_OF_BALANCING_ROUNDS_METRIC_NAME,
+            "Total number of balancing rounds",
+            "unit"
+        );
+        this.shardMovesCounter = meterRegistry.registerLongCounter(
+            NUMBER_OF_SHARD_MOVES_METRIC_NAME,
+            "Total number of shard moves",
+            "unit"
+        );
+
+        this.shardMovesHistogram = meterRegistry.registerLongHistogram(
+            NUMBER_OF_SHARD_MOVES_HISTOGRAM_METRIC_NAME,
+            "Number of shard movements executed in a balancing round",
+            "unit"
+        );
+        this.shardCountHistogram = meterRegistry.registerLongHistogram(
+            NUMBER_OF_SHARDS_METRIC_NAME,
+            "change in node shard count per balancing round",
+            "unit"
+        );
+        this.diskUsageHistogram = meterRegistry.registerDoubleHistogram(
+            DISK_USAGE_BYTES_METRIC_NAME,
+            "change in disk usage in bytes per balancing round",
+            "unit"
+        );
+        this.writeLoadHistogram = meterRegistry.registerDoubleHistogram(
+            WRITE_LOAD_METRIC_NAME,
+            "change in write load per balancing round",
+            "1.0"
+        );
+        this.totalWeightHistogram = meterRegistry.registerDoubleHistogram(
+            TOTAL_WEIGHT_METRIC_NAME,
+            "change in total weight per balancing round",
+            "1.0"
+        );
+    }
+
+    public void addBalancingRoundSummary(BalancingRoundSummary summary) {
+        balancingRoundCounter.increment();
+        shardMovesCounter.incrementBy(summary.numberOfShardsToMove());
+        shardMovesHistogram.record(summary.numberOfShardsToMove());
+
+        for (Map.Entry<DiscoveryNode, NodesWeightsChanges> changesEntry : summary.nodeToWeightChanges().entrySet()) {
+            DiscoveryNode node = changesEntry.getKey();
+            NodesWeightsChanges weightChanges = changesEntry.getValue();
+            BalancingRoundSummary.NodeWeightsDiff weightsDiff = weightChanges.weightsDiff();
+
+            // Math.abs on a long value does not have a corresponding positive value for long.MIN_VALUE.
+            // This is impossible here, so multiply by negative one if negative
+            long shardCountDiff = weightsDiff.shardCountDiff();
+            if (shardCountDiff < 0) {
+                shardCountDiff *= -1;
+            }
```

Comment:
I'm not sure I follow the reasoning here? can't we just use Math.abs if we know the diff won't be `MIN_VALUE`?